game/tests.py

```
@@ -1180,6 +1180,75 @@ def test_get_ai_move_falls_back_to_engine_when_book_empty(self):
+        ChessGame._opening_book = None
+        with mock.patch.object(ChessGame, 'OPENING_BOOK_PATH', '/nonexistent/bogus.json'):
+            with mock.patch('builtins.open', mock.mock_open(read_data='[]')):
+                book = ChessGame._load_opening_book()
+        self.assertEqual(book, {})
+        ChessGame._opening_book = None
+
+    def test_load_rejects_list_root_file(self):
+        """A JSON file containing a list (not dict) must produce empty dict."""
+        ChessGame._opening_book = None
+        with mock.patch('builtins.open', mock.mock_open(read_data='[]')):
+            with mock.patch.object(ChessGame, 'OPENING_BOOK_PATH', '/fake/book.json'):
```

Ruff (0.15.14)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)